### PR TITLE
[FIX] website: Never find pages from other websites

### PR DESCRIPTION
**Pull request by yajo**
Steps to reproduce:

1. Go to "My Website".
1. Click *New > New Page*.
1. Page title: "hello", add to menu: yes.
1. Add a snippet, write "hello 1", save.
1. Go to "My Website 2".
1. Click *New > New Page*.
1. Page title: "hello", add to menu: yes.
1. Add a snippet, write "hello 2", save.
1. Go to *Pages > Edit Menu*.
1. Edit the "hello" menu.
1. Change URL to `/hello-world` and save.
1. Click on the "hello" menu.
1. A page with "hello 2" appears.
1. Go to "My Website".
1. Go to *Pages > Edit Menu*.
1. Edit the "hello" menu.
1. Change URL to `/hello-world` and save.
1. Click on the "hello" menu.

Current behavior: Error 404

Expected behavior: A page with "hello 1" appears.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@JKE-be @rdeodoo can you please check this?

**Comment by pedrobaeza** [comment]
@Yajo I suppose this has to do with view auto-removal, isn't it?

**Comment by Yenthe666** [comment]
LGTM 👍

File: addons/website/models/website.py [hunk 1]
```diff
@@ -1311,7 +1311,7 @@ def replace_id(old_id, new_id):
                 if menu_id.page_id:
                     menu_id.page_id = None
             else:
-                page = self.env['website.page'].search(['|', ('url', '=', menu['url']), ('url', '=', '/' + menu['url'])], limit=1)
+                page = self.env['website.page'].search(['|', ('url', '=', menu['url']), ('url', '=', '/' + menu['url']), '|', ('website_id', '=', False), ('website_id', '=', website_id)], limit=1)
```
Comment (mreficent) [inline addons/website/models/website.py]:
```suggestion
                page = self.env['website.page'].search(['|', ('url', '=', menu['url']), ('url', '=', '/' + menu['url']), ('website_id', 'in', [website_id, False])], limit=1)
```

**Comment by Yajo** [comment]
Fixed.

**Comment by pedrobaeza** [comment]
Great, which one do +r?

**Comment by JKE-be** [comment]
@robodoo r+

**Pull request closed without merging**